### PR TITLE
Use GCR registry mirror in Travis for Linux-based platforms

### PR DESCRIPTION
**Pull request by nkubala**
this configures a Dockerhub registry mirror in some of our Travis jobs so we don't hit the Dockerhub rate limit (as often anyway).

this uses a bit of a hack with `jq` to force update Travis' `daemon.json` file since this is the only way I could get it to take. this means we can only use it on Linux-based platforms.

### CODE REVIEW

**Comment by MarlonGamez** [comment]
@nkubala could we install `jq` on the MacOS platforms?

**Comment by nkubala** [comment]
> @nkubala could we install `jq` on the MacOS platforms?

sorry my description wasn't really accurate. we **can** install it on Windows/OS X, but the problem is that Travis doesn't support platform-specific `before_script` clauses. to do this we'd need to have the script itself check the platform and conditionally use different methods to install the tool - very annoying to do in bash, can be done using a separate small binary which does this (we could write in `go` or whatever) but i didn't want to do that here :)

**Comment by MarlonGamez** [comment]
Ah okay, I think we can track doing this for MacOS and Windows as follow up work then.

This might not be too bad to do in bash actually, looks like travis gives us `$TRAVIS_OS_NAME` according to https://docs.travis-ci.com/user/multi-os/ which could make things easier. Hopefully we can try this out in a follow-up!

**Review by MarlonGamez — APPROVED** [review]
Nice!

File: .travis.yml [hunk 1]
```diff
@@ -13,6 +13,15 @@ branches:
 git:
   submodules: false
 
+_registry_mirror: &registry_mirror
+  before_script:
+    - tmpdaemon=$(mktemp)
+    - sudo jq '."registry-mirrors" += ["https://mirror.gcr.io"]' /etc/docker/daemon.json > $tmpdaemon
```
Comment (briandealwis) [inline .travis.yml:19]: Ooo that's pretty cool.  jq feels like black magic.

**Pull request merged**